config/remote/example_test.go

```diff
@@ -20,13 +21,17 @@ func Example() {
 	}
 	key := "core.yaml"
 	envEtcdAddrs := strings.Split(addr, ",")
-	cfg := clientv3.Config{
+	_ = put(clientv3.Config{
 		Endpoints:   envEtcdAddrs,
-		DialTimeout: time.Second,
-	}
-	_ = put(cfg, key, "name: etcd")
+		DialTimeout: 5 * time.Second,
+	}, key, "name: etcd")
 
-	c := core.New(etcd.WithKey(cfg, key, yaml.Codec{}))
+	cfg := remote.Config{
+		Endpoints: envEtcdAddrs,
+		Name:      "etcd",
+		Key:       key,
+	}
+	c := core.New(remote.WithKey(cfg, yaml.Codec{}))
```

Comment:
why change the signature of WithKey?

Reply:
Thank you for your review.
- I agree with you on `viper`. The original `crypt` is not very good. I have rebuilt the code of his `backend` part, and added related tests. Maybe I should write the `backend` part in the core?
- Add `Name`( for choice backend client ) and `WatchInterval` ( for timer selector of Watch). A little too many parameters. So I packed them into `remote.Config`.